example_notebooks/cube_4d.ipynb

```diff
@@ -0,0 +1,260 @@
+    "        self._viewer.set_image(image)\n",
+    "        \n",
+    "    def _mouse_click_cb(self, viewer, event, data_x, data_y):\n",
+    "        super()._mouse_click_cb(viewer, event, data_x, data_y)\n",
+    "        \n",
+    "        # Plot line profile\n",
+    "        image = self._viewer.get_image()\n",
+    "        if image is not None and self.line_plot is not None:\n",
+    "            mddata = image.get_mddata()\n",
+    "            ix = int(round(data_x))\n",
+    "            iy = int(round(data_y))\n",
+    "            with self.line_out:\n",
+    "                self.line_plot.clear()\n",
+    "                self.line_plot.plot(mddata[self._4d_idx, :, iy, ix])\n",
+    "                clear_output(wait=True)\n",
+    "                display(self.line_plot.figure)\n",
+    "\n",
+    "    def show_slice(self, n):\n",
+    "        image = self._viewer.get_image()\n",
+    "        image.set_naxispath([n, self._4d_idx])\n",
+    "        self._viewer.redraw(whence=0)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "logger = get_logger('my viewer', log_stderr=True, log_file=None, level=30)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "w = CubeWidget(logger=logger)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Generate a fake 4D data cube."
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {},
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "pf = None  # Not using FITS file pointer\n",
+    "cube4d = np.random.random((2, 5, 100, 200))\n",
+    "n_3d = cube4d.shape[1]\n",
+    "\n",
+    "# Convert to NDData.\n",
+    "im = NDData(cube4d, wcs=WCS(naxis=4))\n",
+    "\n",
+    "# Loading a fake WCS into widget sometimes gives error. Try reload before giving up.\n",
+    "w.load_nddata(im)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**--- OR ---**\n",
+    "\n",
+    "Load an existing data cube."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import dask.array as da\n",
+    "from astropy.io import fits\n",
+    "\n",
+    "# This is a larger-than-memory cube that is 43GB.\n",
+    "filename = '/redkeep/dungeons/tess_big_cubes/tess-s0001-1-1-cube.fits'\n",
+    "pf = fits.open(filename, memmap=True)\n",
+    "im = da.from_array(pf[1].data, name='mytesscube')\n",
+    "im = im.T  # TESS cube is transposed by ASB\n",
+    "# Shape now is (2, 1282, 2136, 2078)\n",
+    "n_3d = im.shape[1]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# This needs https://github.com/ejeschke/ginga/pull/805\n",
+    "# TODO: Why is it taking so long?\n",
+    "w.load_array(im)"
```

Comment:
@ejeschke , `w.load_array(im)` here calls `EnhancedCanvasView.load_data()` on the Dask array. It takes way too long is locking up the computer a little. I am using your PR ejeschke/ginga#805. Is this not the correct way to load a Dask array into the widget viewer?